openwisp_firmware_upgrader/tests/test_signals.py

```diff
@@ -0,0 +1,141 @@
+import os
+from unittest.mock import patch
+
+from django.conf import settings
+from django.test import TestCase, TransactionTestCase
+from swapper import load_model
+
+from ..tasks import delete_firmware_files
+from .base import TestUpgraderMixin
+
+Build = load_model('firmware_upgrader', 'Build')
+Category = load_model('firmware_upgrader', 'Category')
+FirmwareImage = load_model('firmware_upgrader', 'FirmwareImage')
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class TestSignals(TestUpgraderMixin, TestCase):
+    def setUp(self):
+        # Ensure PRIVATE_STORAGE_ROOT exists
+        os.makedirs(settings.PRIVATE_STORAGE_ROOT, exist_ok=True)
+        self._create_test_files()
+
+    def tearDown(self):
+        self._remove_test_files()
+        super().tearDown()
+
+    def _create_test_files(self):
+        os.makedirs(settings.PRIVATE_STORAGE_ROOT, exist_ok=True)
+        with open(self.FAKE_IMAGE_PATH, 'wb') as f:
+            f.write(b'fake firmware image file')
+        with open(self.FAKE_IMAGE_PATH2, 'wb') as f:
+            f.write(b'fake firmware image file 2')
+
+    def _remove_test_files(self):
+        for path in [self.FAKE_IMAGE_PATH, self.FAKE_IMAGE_PATH2]:
+            try:
+                os.remove(path)
+            except FileNotFoundError:
+                pass
+        try:
+            os.rmdir(settings.PRIVATE_STORAGE_ROOT)
+        except (FileNotFoundError, OSError):
+            pass
+
+    def test_build_delete_files(self):
+        build = self._create_build()
+        fw = self._create_firmware_image(build=build)
+        file_path = fw.file.path
+        self.assertTrue(os.path.exists(file_path))
+
+        with patch(
+            'openwisp_firmware_upgrader.tasks.delete_firmware_files.delay'
+        ) as mock:
+            build.delete()
+            mock.assert_called_once()
+            file_paths = mock.call_args[0][0]
+            self.assertEqual(len(file_paths), 1)
+            self.assertEqual(file_paths[0], fw.file.name)
```

Comment:
Let's not mock these. Instead, we should verify that deleting the `build` deletes these files.